src/azul/terra.py

```diff
@@ -150,9 +150,10 @@ def register_with_sam(self) -> None:
             log.info('Google service account previously registered with SAM.')
         elif (response.status == 500
               and 'Cannot update googleSubjectId' in response.data.decode()):
-            log.warning('Unable to register %s. SAM does not allow re-registration of '
-                        'service account emails. Refer to the troubleshooting section '
-                        'of the README.', self.credentials.service_account_email)
+            raise RequirementError('Unable to register %s. SAM does not allow '
```

Comment:
I like that the script is exited if this is hit, there appears to be a problem with the interpolation of the email into the text.

```
Apply complete! Resources: 3 added, 11 changed, 2 destroyed.
/Applications/Xcode.app/Contents/Developer/usr/bin/make link_health_checks
python /Users/amar/dev/hca/azul/scripts/manage_health_checks.py --link
2020-11-12 09:11:35,375 INFO    MainThread: Found credentials in environment variables.
2020-11-12 09:11:36,323 INFO    MainThread: DCP wide health check does not exist for amar
/Applications/Xcode.app/Contents/Developer/usr/bin/make grafana
python /Users/amar/dev/hca/azul/scripts/update_grafana.py
2020-11-12 09:11:36,655 INFO    MainThread: Skipping publishing of Grafana dashboard
/Applications/Xcode.app/Contents/Developer/usr/bin/make sam
python /Users/amar/dev/hca/azul/scripts/register_sam.py
2020-11-12 09:11:38,591 INFO    MainThread: Found credentials in environment variables.
Traceback (most recent call last):
  File "/Users/amar/dev/hca/azul/scripts/register_sam.py", line 37, in <module>
    main()
  File "/Users/amar/dev/hca/azul/scripts/register_sam.py", line 23, in main
    tdr.register_with_sam()
  File "/Users/amar/dev/hca/azul/src/azul/terra.py", line 153, in register_with_sam
    raise RequirementError('Unable to register %s. SAM does not allow '
azul.RequirementError: ('Unable to register %s. SAM does not allow re-registration of service account emails. Refer to the troubleshooting section of the README.', 'azul-ucsc-2-amar@human-cell-atlas-travis-test.iam.gserviceaccount.com')
make[2]: *** [sam] Error 1
make[1]: *** [apply] Error 2
make: *** [deploy] Error 2
```